ui/src/app/pages/workspace/workspace-edit.spec.tsx

```diff
@@ -687,6 +748,48 @@ describe('WorkspaceEdit', () => {
     expect(mockNavigate).toHaveBeenCalledTimes(1);
   });
 
+  // regression test for RW-5132 - async version
+  it('prevents multiple (async) Workspace creations via the same confirmation dialog', async () => {
+    environment.enableAsyncWorkspaceOperations = true;
+    workspaceEditMode = WorkspaceEditMode.Duplicate;
+    const wrapper = component();
+    await waitOneTickAndUpdate(wrapper);
+
+    wrapper
+      .find('[data-test-id="review-request-btn-false"]')
+      .first()
+      .simulate('click');
+    await waitOneTickAndUpdate(wrapper);
+
+    const numBefore = workspacesApi.workspaceOperations.length;
+    wrapper
+      .find('[data-test-id="workspace-save-btn"]')
+      .first()
+      .simulate('click');
+    await waitOneTickAndUpdate(wrapper);
+
+    wrapper
+      .find('[data-test-id="workspace-confirm-save-btn"]')
+      .first()
+      .simulate('click');
+    await waitOneTickAndUpdate(wrapper);
+
+    wrapper
+      .find('[data-test-id="workspace-confirm-save-btn"]')
+      .first()
+      .simulate('click');
+    await waitOneTickAndUpdate(wrapper);
+
+    wrapper
+      .find('[data-test-id="workspace-confirm-save-btn"]')
+      .first()
+      .simulate('click');
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(workspacesApi.workspaceOperations.length).toEqual(numBefore + 1);
+    expect(mockNavigate).toHaveBeenCalledTimes(1);
```

Comment:
Is there an error or warning message if a user tries to create more than one workspace?

Reply:
the fix was to make it no longer possible to do so: the button is disabled after clicking